src/themes/themesController.ts

```diff
@@ -0,0 +1,480 @@
+        }
+        if (!srcTheme["primary-button-focus-border-color"]) {
+            destTheme["primary-button-focus-border-color"] = srcTheme["primary-button-color"];
+        }
+        if (!srcTheme["radio-button-hover-border-color"]) {
+            destTheme["radio-button-hover-border-color"] = srcTheme["primary-button-color"];
+        }
+
+        // determined by "text-color"
+        if (!srcTheme["secondary-button-text-color"]) {
+            destTheme["secondary-button-text-color"] = srcTheme["text-color"];
+        }
+        if (!srcTheme["secondary-button-disabled-text-color"]) {
+            destTheme["secondary-button-disabled-text-color"] = srcTheme["text-color"];
+        }
+        if (!srcTheme["radio-button-border-color"]) {
+            destTheme["radio-button-border-color"] = srcTheme["text-color"];
+        }
+        if (!srcTheme["radio-button-checked-background-color"]) {
+            destTheme["radio-button-checked-background-color"] = srcTheme["text-color"];
+        }
+        if (!srcTheme["radio-button-hover-background-color"]) {
+            let provided: string = srcTheme["text-color"];
+            this.setMissingColorFromAnotherProperty("radio-button-hover-background-color", provided, 0.8, destTheme);
+        }
+        if (!srcTheme["radio-button-disabled-color"]) {
+            let provided: string = srcTheme["text-color"];
+            this.setMissingColorFromAnotherProperty("radio-button-disabled-color", provided, 0.2, destTheme);
+        }
+        if (!srcTheme["radio-button-disabled-border-color"]) {
+            let provided: string = srcTheme["text-color"];
+            this.setMissingColorFromAnotherProperty("radio-button-disabled-border-color", provided, 0.2, destTheme);
+        }
+
+        // determined by "secondary-button-color"
+        if (!srcTheme["secondary-button-hover-color"]) {
+            destTheme["secondary-button-hover-color"] = srcTheme["secondary-button-color"];
+        }
+
+        // determined by "secondary-button-disabled-color"
+        if (!srcTheme["secondary-button-disabled-border"]) {
+            destTheme["secondary-button-disabled-border"] = "1px solid " + srcTheme["secondary-button-disabled-color"];
+        }
+        
+        // determined by "secondary-button-border"
+        if (!srcTheme["secondary-button-hover-border"]) {
+            destTheme["secondary-button-hover-border"] = srcTheme["secondary-button-border"];
+        }
+        if (!srcTheme["secondary-button-focus-border-color"]) {
+            let secondaryBtnBorderElement: string[] = srcTheme["secondary-button-border"].split(" ");
+            destTheme["secondary-button-focus-border-color"] = secondaryBtnBorderElement[secondaryBtnBorderElement.length - 1];
+        }
+
+    }
+
+    /**
+     * Create theme elements for applying the theme
+     */
+    public static createThemeStyle(): void {
+        let themesStyles = document.createElement('style');
```

Comment:
There are several issues with this approach:
1. There's no reason to create empty styles first and to replace them later
2. Every time you modify innerHTML browser will try to apply all your changes and to render them. So this is an expensive operation. And you do it around 30 times on every them apply. And this will become a very expensive operation
3. Replacing of styles finding styles with `indexOf` can introduce lots of errors and difficult to understand.

I suggest you do this approach:
```
let newStyles = '';
newStyles+= `.${ styles.hyperLinkTheme } {color: #fff}`;
newStyles+= `...`;
...
document.getElementById('ms-consent-banner-theme-styles').innerHTML = newStyles;
```